src/ddmd/traits.d

```diff
@@ -917,27 +917,34 @@ extern (C++) Expression semanticTraits(TraitsExp e, Scope* sc)
         if (dim != 1)
             return dimError(1);
 
+        LINK link;
         auto o = (*e.args)[0];
-        auto s = getDsymbol(o);
-        Declaration d;
-        if (!s || (d = s.isDeclaration()) is null)
+        auto t = isType(o);
+        TypeFunction tf = null;
+        if (t)
         {
-            e.error("argument to `__traits(getLinkage, %s)` is not a declaration", o.toChars());
-            return new ErrorExp();
+            if (t.ty == Tfunction)
+                tf = cast(TypeFunction)t;
+            else if (t.ty == Tdelegate)
+                tf = cast(TypeFunction)t.nextOf();
+            else if (t.ty == Tpointer && t.nextOf().ty == Tfunction)
+                tf = cast(TypeFunction)t.nextOf();
```

Comment:
The `assert` will trip if someone passes a non-function type to the trait, so this change will not work correctly.